src/client.ts

```diff
@@ -177,6 +189,59 @@ export class SubscriptionClient {
     });
   }
 
+  public applyMiddlewares(options: SubscriptionOptions): Promise<SubscriptionOptions> {
+    return new Promise((resolve, reject) => {
+      const queue = (funcs: MiddlewareInterface[], scope: any) => {
+        const next = () => {
```

Comment:
Following lines should be change to:
```
      const queue = (funcs: MiddlewareInterface[], scope: any) => {
        const next = (error) => {
          if (error) {
            reject(error);
          } else {
            if (funcs.length > 0) {
              const f = funcs.shift();
              if (f) {
                f.applyMiddleware.apply(scope, [options, next]);
              }
            } else {
              resolve(options);
            }
          }
        };
        next();
      };
```

And [this](https://github.com/srtucker22/subscriptions-transport-ws/blob/b5756399525aed21ef355aa04c5870ccd3d4b6ff/src/client.ts#L149) line update to:
```
}).catch(e => handler([e]));
```